channeldb/payments.go

```diff
@@ -601,9 +604,28 @@ func (d *DB) QueryPayments(query PaymentsQuery) (PaymentsResponse, error) {
 				totalPayments uint64
 				err           error
 			)
-			countFn := func(_, _ []byte) error {
-				totalPayments++
 
+			countFn := func(sequenceKey, hash []byte) error {
+				r := bytes.NewReader(hash)
+				paymentHash, err := deserializePaymentIndex(r)
```

Comment:
Since counting payments is already slow, this would make it even slower - for nodes with millions of payments this would take a very long time. I think at least we should make this query conditional, that,
1. if there aren't any date restrictions, use the original `totalPayments++` approach.
2. if there is a date range, use this `countFn`.

But I still think this would be very slow, cc @bhandras for some help here, maybe we shouldn't touch it until we can find an efficient counting method?

Reply:
Maybe there is a way to measure the performance impact and only move forward if it is near the same of we have today without the fix.